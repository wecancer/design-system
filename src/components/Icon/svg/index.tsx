import ArrowLeftSvg from './arrow-left.svg'
import ArrowDownSvg from './arrow-down.svg'
import ChartTreeMap from './chart-treemap.svg'
import ChartBar from './chart-bar.svg'
import CalendarSvg from './calendar.svg'
import CheckSvg from './check.svg'
import CloseSvg from './close.svg'
import LongArrowRightSvg from './long-arrow-right.svg'
import MinusSvg from './minus.svg'
import PlugSvg from './plus.svg'
import SearchSvg from './search.svg'
import SpinnerDots from './spinner-dots.svg'
import SpinnerSolid from './spinner-solid.svg'
import TimesSolidSvg from './times-solid.svg'
import TimesSvg from './times.svg'
import UnviewSvg from './unview.svg'
import ViewSvg from './view.svg'

export default {
  arrowLeft: ArrowLeftSvg,
  arrowDown: ArrowDownSvg,
  chartBar: ChartBar,
  chartTreeMap: ChartTreeMap,
  calendar: CalendarSvg,
  check: CheckSvg,
  close: CloseSvg,
  longArrowRight: LongArrowRightSvg,
  minus: MinusSvg,
  plus: PlugSvg,
  search: SearchSvg,
  spinnerDots: SpinnerDots,
  spinnerSolid: SpinnerSolid,
  times: TimesSvg,
  timesSolid: TimesSolidSvg,
  unview: UnviewSvg,
  view: ViewSvg,
}
