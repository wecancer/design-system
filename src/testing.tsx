import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
