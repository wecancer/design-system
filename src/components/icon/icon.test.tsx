import Icon, { iconsMap, IconsTypes } from '.'
import { render } from '../../testing'

describe('<Icon />', () => {
  it('should render snapshot correctly', () => {
    const { container } = render(
      <>
        {Object.keys(iconsMap).map((iconName) => (
          <Icon key={iconName} name={iconName as IconsTypes} />
        ))}
        <Icon name="plus" size={20} />
        <Icon name="check" className="someclass" />
      </>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render all icons from library', () => {
    const { container } = render(
      <div className="container">
        {Object.keys(iconsMap).map((iconName) => (
          <Icon key={iconName} name={iconName as IconsTypes} />
        ))}
      </div>,
    )
    const items = container.querySelector('.container')

    expect(items?.children).toHaveLength(Object.keys(iconsMap).length)
  })
})
