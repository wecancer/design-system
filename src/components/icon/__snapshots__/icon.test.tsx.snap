// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render snapshot correctly 1`] = `
<div
  data-icon-name="alertInfo"
  style="display: inline-flex; align-items: center; justify-content: center;"
>
  <svg
    fill="none"
    height="16px"
    viewBox="0 0 24 24"
    width="16px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="11"
      stroke="currentColor"
      stroke-width="2"
    />
    <path
      d="M12 7V12"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
    <path
      d="M12 16V16.5"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;
